### PR TITLE
Remove examples that have been moved to HDF5Examples folder

### PR DESCRIPTION
**Pull request by byrnHDF**

### CODE REVIEW

**Review by lrknox — APPROVED** [review]

File: HDF5Examples/C/H5PAR/ph5example.c
Comment (jhendersonHDF) [inline HDF5Examples/C/H5PAR/ph5example.c:1]: Looks like this file was moved under `HDF5Examples/C/H5PAR/`, but does the build code there take this file into account?

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -32,28 +32,8 @@ set (examples
...
 )
 
-if (H5_HAVE_PARALLEL)
-  set (parallel_examples
-    ph5example
-    ph5_filtered_writes
-    ph5_filtered_writes_no_sel
-  )
-
-  if (HDF5_ENABLE_SUBFILING_VFD)
```
Comment (jhendersonHDF) [inline examples/CMakeLists.txt:52]: Looks like this particular piece is missing from the CMakeLists.txt file in the HDF5Examples directory

File: examples/CMakeTests.cmake [hunk 2]
```diff
@@ -30,8 +30,6 @@ set (test_ex_CLEANFILES
     group.h5
     groups.h5
     hard_link.h5
-    h5_subfiling_default_example.h5
-    h5_subfiling_custom_example.h5
```
Comment (jhendersonHDF) [inline examples/CMakeTests.cmake:34]: Do the examples under HDF5Examples not have a "clean files" list anymore?

File: HDF5Examples/C/H5PAR/ph5example.c
Comment (byrnHDF) [reply HDF5Examples/C/H5PAR/ph5example.c:1]: No - I need to ask for help on verifying the code. I moved it to work on it.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (byrnHDF) [reply examples/CMakeLists.txt:52]: I thought it was already in the PAR directory.

File: examples/CMakeTests.cmake (hunk 2 above)
Comment (byrnHDF) [reply examples/CMakeTests.cmake:34]: Need to check.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (jhendersonHDF) [reply examples/CMakeLists.txt:52]: Oh I see it's in C_sourcefiles.cmake instead

**Review by jhendersonHDF — APPROVED** [review]

**Pull request merged**